platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/telemetry/MapboxEventManager.java

```diff
@@ -551,7 +553,16 @@ protected Void doInBackground(Void... voids) {
                     jsonObject.putOpt(MapboxEvent.ATTRIBUTE_SOURCE, evt.get(MapboxEvent.ATTRIBUTE_SOURCE));
                     jsonObject.putOpt(MapboxEvent.ATTRIBUTE_SESSION_ID, evt.get(MapboxEvent.ATTRIBUTE_SESSION_ID));
                     jsonObject.putOpt(MapboxEvent.KEY_LATITUDE, evt.get(MapboxEvent.KEY_LATITUDE));
-                    jsonObject.putOpt(MapboxEvent.KEY_LONGITUDE, evt.get(MapboxEvent.KEY_LONGITUDE));
```

Comment:
Normally I'd agree that conditional wrapping should be avoided, however in this case it's needed.  Reason being is that Latitude and Longitude data can also come from other events besides `location`.  Specifically gesture events like [`map.click`](https://github.com/mapbox/mapbox-gl-native/blob/b864fd4ea543b85daf48380fc36388f8cedf0032/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java#L1391) and [`map.dragend`](https://github.com/mapbox/mapbox-gl-native/blob/b864fd4ea543b85daf48380fc36388f8cedf0032/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java#L1412).  This is handled automatically via `json.putOpt()` for Latitude because it's value (if one exists) can always be used as is.